ui/app/AppLayouts/Chat/components/ProfilePopup.qml

```diff
@@ -7,62 +7,195 @@ import "./"
+          readOnly: true
+          color: Theme.darkGrey
+          font.family: "Inter"
+      }
+
+      // TODO(pascal): implement qrcode view
+      // Rectangle {
+      //     id: qrCodeButton
+      //     height: 32
+      //     width: 32
+      //     anchors.top: parent.top
+      //     anchors.topMargin: Theme.padding
+      //     anchors.right: parent.right
+      //     anchors.rightMargin: 32 + Theme.smallPadding
+      //     radius: 8
+
+      //     Image {
+      //         source: "../../../../shared/img/qr-code-icon.svg"
+      //         anchors.horizontalCenter: parent.horizontalCenter
+      //         anchors.verticalCenter: parent.verticalCenter
+      //     }
+
+      //     MouseArea {
+      //         cursorShape: Qt.PointingHandCursor
+      //         anchors.fill: parent
+      //         hoverEnabled: true
+      //         onExited: {
+      //             qrCodeButton.color = Theme.white
+      //         }
+      //         onEntered:{
+      //             qrCodeButton.color = Theme.grey
+      //         }
+      //     }
+      // }
+    }
+
+    Text {
+      id: labelEnsUsername
+      text: qsTr("ENS username")
+      font.pixelSize: 13
+      font.weight: Font.Medium
+      color: Theme.darkGrey
+      anchors.left: parent.left
+      anchors.leftMargin: Theme.smallPadding
       anchors.top: parent.top
+      anchors.topMargin: Theme.smallPadding
+    }
+
+    Text {
+      id: valueEnsName
+      text: "@emily.stateofus.eth"
+      font.pixelSize: 14
       anchors.left: parent.left
-      font.bold: true
-      font.pixelSize: 17
-      anchors.leftMargin: 16
-      anchors.topMargin: Theme.padding
-      anchors.bottomMargin: Theme.padding
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: labelEnsUsername.bottom
+      anchors.topMargin: Theme.smallPadding
     }
 
-    Rectangle {
-      id: profilePic
-      width: 120
-      height: 120
-      radius: 100
-      border.color: "#10000000"
-      border.width: 1
-      color: Theme.transparent
-      anchors.horizontalCenter: parent.horizontalCenter
-      anchors.top: parent.top
-      anchors.topMargin: 16
-      Image {
-          width: 120
-          height: 120
-          fillMode: Image.PreserveAspectFit
-          source: identicon
-      }
+    Text {
+      id: labelChatKey
+      text: qsTr("Chat key")
+      font.pixelSize: 13
+      font.weight: Font.Medium
+      color: Theme.darkGrey
+      anchors.left: parent.left
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: valueEnsName.bottom
+      anchors.topMargin: Theme.padding
     }
 
     Text {
-      id: userNameText
-      text: userName
-      anchors.top: profilePic.bottom
-      anchors.topMargin: 16
-      anchors.horizontalCenter: parent.horizontalCenter
-      font.bold: true
-      font.pixelSize: 16
+      id: valueChatKey
+      text: fromAuthor.substr(0, 14) + "..." + fromAuthor.substr(fromAuthor.length - 14)
```

Comment:
Instead of eliding yourself, you can use the QML property called `elide`. You just have to give the Text an anchor on the right and the left (or a width), then you set `elide` to `center` (or `middle`, I can't remember) and QML will do the `...` on itself

Reply:
Ah thanks will check that out!

Reply:
updated, cool stuff